decidim/decidim
# Add the trustee zone for users

#### :tophat: What? Why?
A user can act as a trustee for an election.

#### :pushpin: Related Issues
- Fixes #6360
- Related to #6535

#### Testing
After (adding a user as a trustee for a participatory space)[https://github.com/decidim/decidim/pull/6535], login with that user and visit the user account page. A new "Trustee zone" link should appear in the user profile menu.

#### :clipboard: Checklist
:rotating_light: Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.


### :camera: Screenshots
* Identification keys generation

![imagen](https://user-images.githubusercontent.com/453545/96492733-5ea6d500-1244-11eb-93bb-2a81cd2f292d.png)

* After generating identification keys (a "Save file" dialog will be open too)

![imagen](https://user-images.githubusercontent.com/453545/96492770-6b2b2d80-1244-11eb-9ec8-c133fb3efe15.png)

* If identification keys are not present on the user browser

![imagen](https://user-images.githubusercontent.com/453545/96492884-931a9100-1244-11eb-9cec-17eea8a073bc.png)


### ⏯️ Steps to test (and re-test)

To try the Trustee Zone:
1. Login with the admin@example.org user
1. Go to the admin zone
1. Go to a participatory space with at least one election component
1. Go to the Trustees admin zone
1. Add the current user as a Trustee for the participatory space
1. Open a new window and go to the account page for the user
1. Click on the Trustee zone link, below the Delete account link
1. After generating and submitting the identification keys, the public key for the trustee should appear in the admin zone, in the list of trustees for the participatory space.

To reset the Trustee public key:
1. Go to the admin zone for the participatory space with the trustee added
1. Remove the user admin@example.org from the trustees list. Ensure the user is not a trustee on any other space.
1. Add the current user again as a Trustee.
1. The public key should be empty. 




:hearts: Thank you!
